src/Microsoft.ML.AutoML/API/ExperimentBase.cs

```diff
@@ -194,7 +194,8 @@ public ExperimentResult<TMetrics> Execute(IDataView trainData, IDataView validat
             IProgress<CrossValidationRunDetail<TMetrics>> progressHandler = null)
         {
             UserInputValidationUtil.ValidateNumberOfCVFoldsArg(numberOfCVFolds);
-            var splitResult = SplitUtil.CrossValSplit(Context, trainData, numberOfCVFolds, columnInformation?.SamplingKeyColumnName);
+            UserInputValidationUtil.ValidateSamplingKey(columnInformation?.SamplingKeyColumnName, columnInformation?.GroupIdColumnName, _task);
+            var splitResult = SplitUtil.CrossValSplit(Context, trainData, numberOfCVFolds, columnInformation?.GroupIdColumnName);
```

Comment:
Should we add the `ValidateSamplingKey` also to the other `Execute` overloads that use `columnInformation.SamplingKeyColumn` to split data?

For example this overload:
https://github.com/dotnet/machinelearning/blob/e8fa7310f936edb3f048a9ac3d0ef1f3a8f5b6f6/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L96-L116

That overload uses CVSplit if the data is over 15000 rows. But notice that it also splits the data using TrainTestSplit if it is under that number of rows, and TrainTestSplit also uses `samplingKeyColumn`; the `samplingKeyColumn` for TTSplit should also be the GroupId column if it's done for ranking, for the same reasons we've already discussed offline.

Reply:
(unresolving) I see you've updated other `Execute` overloads addressing this comment, but I think you missed one overload? The one at:
[https://github.com/antoniovs1029/machinelearning/blob/bb13d629000c218136e741b643767cf45ae12fc4/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L221-L232](https://github.com/antoniovs1029/machinelearning/blob/bb13d629000c218136e741b643767cf45ae12fc4/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L221-L232)

---
In reply to: [451036924](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451036924) [](ancestors = 451036924)

Reply:
BTW, just curious, do you happen to know which `Execute` method does ModelBuilder calls? /cc: @LittleLittleCloud

---
In reply to: [451951124](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451951124) [](ancestors = 451951124,451036924)